eiffellib/events/eiffel_base_event.py

```diff
@@ -169,14 +169,31 @@ class EiffelBaseEvent(object):
 
     schema_file = None
     __schema = None
+    __routing_key = "eiffel.{family}.{type}.{tag}.{domain_id}"
+    family = None
+    tag = None
+    domain_id = None
     version = "0.0.1"
     meta = EiffelBaseMeta("EiffelBaseEvent", version)
     links = EiffelBaseLink()
 
-    def __init__(self, version=None):
-        """Initialize with a base data object."""
+    def __init__(self, version=None, family="_", tag="_", domain_id="_"):
+        """Initialize with a base data object.
+
+        :param version: If not None use this version when loading json schemas.
+        :type version: str
+        :param family: Routing key family as per the sepia recommendation. Defaults to "_".
+        :type family: str
+        :param tag: Routing key tag as per the sepia recommendation. Defaults to "_".
+        :type tag: str
+        :param domain_id: Routing key domain id as per the sepia recommendation. Defaults to "_".
+        :type domain_id: str
+        """
         if version is not None:
             self.version = version
+        self.family = family
+        self.tag = tag
+        self.domain_id = domain_id
```

Comment:
Should this also affect `meta.source.domainId`?

Reply:
I don't know.. Should we also add `meta.source.domainId` to this automatically in that case?
Do we want implicit behavior?

Reply:
Implicit isn't bad if documented :-) I'd say that `meta.source.domainId` and the domain token in the routing key should match each other in the normal case. As a library user I'd be annoyed if I'd have to pass the domain in the constructor but _also_ modify the `meta` object by hand.

Reply:
We will now add `event.domain_id` to `meta.source.domainId` if it's set.
It will also add `RabbitMQPublisher.source.domainId` to `event.domain_id` if `event.domain_id` is the default value of "_".

I.e. `event.domain_id` will always take precedence over `RabbitMQPublisher.source.domainId` if set